python/generateTemplate.py

```diff
@@ -0,0 +1,1155 @@
+                refract_end = diff_refraction(wavelength=wl_end, wavelength_ref=wavelength_midpoint,
+                                              zenith_angle=zenith_angle,
+                                              observatory=observatory, weather=weather)
+                refract_start_pixels = refract_start.asArcseconds()/pixel_scale.asArcseconds()
+                refract_end_pixels = refract_end.asArcseconds()/pixel_scale.asArcseconds()
+                dx = delta(start=refract_start_pixels*np.sin(rotation_angle.asRadians()),
+                           end=refract_end_pixels*np.sin(rotation_angle.asRadians()))
+                dy = delta(start=refract_start_pixels*np.cos(rotation_angle.asRadians()),
+                           end=refract_end_pixels*np.cos(rotation_angle.asRadians()))
+                yield dcr(dx=dx, dy=dy)
+
+    def create_exposure(self, array, elevation, azimuth, variance=None, era=None, snap=0,
+                        exposureId=0, ra=nanAngle, dec=nanAngle, boresightRotAngle=nanAngle,
+                        weather=lsst_weather, **kwargs):
+        """Convert a numpy array to an LSST exposure, and units of electron counts.
+
+        Parameters
+        ----------
+        array : np.ndarray
+            Numpy array to use as the data for the exposure
+        elevation : lsst.afw.geom Angle
+            Elevation angle of the observation
+        azimuth : lsst.afw.geom Angle
+            Azimuth angle of the observation
+        variance : np.ndarray, optional
+            Numpy array to use as the variance plane of the exposure.
+            If None, the absoulte value of 'array' is used for the variance plane.
+        era : lsst.afw.geom Angle, optional
+            Earth rotation angle (ERA) of the observation.
+            If not set it will be calculated from the latitude, longitude, RA, Dec, and elevation angle
+        snap : int, optional
+            Snap ID to add to the metadata of the exposure. Required to mimic Phosim output.
+        exposureId : int, optional
+            Observation ID of the exposure, a long int.
+        ra : lsst.afw.geom Angle, optional
+            The right ascension of the boresight of the target field.
+        dec : lsst.afw.geom Angle, optional
+            The declination of the boresight of the target field
+        boresightRotAngle : lsst.afw.geom Angle, optional
+            The rotation angle of the field around the boresight.
+        weather : lsst.afw.coord Weather, optional
+            Class containing the measured temperature, pressure, and humidity
+            at the observatory during an observation
+            Weather data is read from the exposure metadata if not supplied.
+        **kwargs :
+            Any additional keyword arguments will be added to the metadata of the exposure.
+
+        Returns
+        -------
+        lsst.afw.image.ExposureD object
```

Comment:
Interesting. One of the first uses of ExposureD for large images that I know of!